### PR TITLE
OFS-165: amount filter for Contract entities

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://openimis.atlassian.net/browse/OFS-165

I want to add "amount" filter for Contract entities in graphQl. "amountFrom" and "amountTo".

### CODE REVIEW

File: contract/gql/gql_types.py [hunk 1]
```diff
@@ -17,9 +18,6 @@ class Meta:
             "id": ["exact"],
             "code": ["exact", "istartswith", "icontains", "iexact"],
             **prefix_filterset("policy_holder__", PolicyHolderGQLType._meta.filter_fields),
-            "amount_notified": ["exact", "lt", "lte", "gt", "gte"],
-            "amount_rectified": ["exact", "lt", "lte", "gt", "gte"],
-            "amount_due": ["exact", "lt", "lte", "gt", "gte"],
             "date_payment_due": ["exact", "lt", "lte", "gt", "gte"],
```
Comment (delcroip) [inline contract/gql/gql_types.py:21]: cannot you leave them even if you don't plan to use them yet ?

File: contract/utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            Q(amount_rectified__gte=amount_from, state__in=[4, 11, 3]) |
+            Q(amount_due__gte=amount_from, state__in=[5, 6, 7, 8, 9, 10])
+        )
+
+    # scenario - both filters set
+    if amount_from and amount_to:
+        return (
+            Q(amount_notified__gte=amount_from, amount_notified__lte=amount_to, state__in=[1, 2]) |
+            Q(amount_rectified__gte=amount_from, amount_rectified__lte=amount_to, state__in=[4, 11, 3]) |
+            Q(amount_due__gte=amount_from, amount_due__lte=amount_to, state__in=[5, 6, 7, 8, 9, 10])
```
Comment (delcroip) [inline contract/utils.py:29]: could you make a array that you reuse of the status ? (like local var)

Comment (sniedzielski) [reply contract/utils.py:29]: Ok, I can do it

File: contract/gql/gql_types.py (hunk 1 above)
Comment (sniedzielski) [reply contract/gql/gql_types.py:21]: Ok. You are right. It should not affect to my changes with new filter "amountFrom/To"

**Comment by sniedzielski** [comment]
@delcroip I've made changes.

**Review by delcroip — APPROVED** [review]

**Pull request merged**